packages/mui-base/src/Select/Select.tsx

```diff
@@ -249,7 +248,7 @@ const Select = React.forwardRef(function Select<
       )}
     </React.Fragment>
   );
-}) as SelectType;
+}) as PolymorphicComponent<SelectTypeMap<any, boolean>>;
```

Comment:
Select is a special case as it has generic props. It should not be cast to `PolymorphicComponent`, but remain `SelectType`. Then, the `SelectType` should be adjusted to match the shape of `PolymorphicComponent` but with a generic parameter.

Reply:
Does this seem correct?

```jsx
export interface SelectType {
  <
    OptionValue extends {},
    RootComponentType extends React.ElementType,
    Multiple extends boolean = false,
  >(
    props: PolymorphicProps<SelectTypeMap<OptionValue, Multiple>, RootComponentType>,
  ): JSX.Element | null;
  propTypes?: any;
  displayName?: string | undefined;
}
```